src/storage/participation.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use iota_client::{
+    node_api::participation::types::{Event, EventId},
+    node_manager::node::Node,
+};
+
+use super::manager::StorageManager;
+use crate::storage::constants::PARTICIPATION_EVENTS;
+
+impl StorageManager {
+    pub(crate) async fn insert_participation_event(
+        &mut self,
+        id: EventId,
+        event: Event,
+        nodes: Vec<Node>,
+    ) -> crate::Result<()> {
+        log::debug!("insert_participation_event {id}");
+
+        let mut events: HashMap<EventId, (Event, Vec<Node>)> = match self.storage.get(PARTICIPATION_EVENTS).await {
+            Ok(events) => serde_json::from_str(&events)?,
+            Err(crate::Error::RecordNotFound(_)) => HashMap::new(),
+            Err(err) => return Err(err),
+        };
+
+        events.insert(id, (event, nodes));
+
+        self.storage.set(PARTICIPATION_EVENTS, &events).await?;
+
+        Ok(())
+    }
+
+    pub(crate) async fn remove_participation_event(&mut self, id: EventId) -> crate::Result<()> {
+        log::debug!("remove_participation_event {id}");
+
+        let mut events: HashMap<EventId, (Event, Vec<Node>)> =
+            serde_json::from_str(&self.storage.get(PARTICIPATION_EVENTS).await?)?;
```

Comment:
Wondering if it should return Ok on RecordNotFound

Reply:
We can do that